pkg/grpcserver/grpcutil.go

```diff
@@ -43,14 +42,40 @@ func Connect(address string, dialOptions []grpc.DialOption) (*grpc.ClientConn, e
 		return nil, err
 	}
 
-	ctx, cancel := context.WithTimeout(context.Background(), time.Minute)
-	defer cancel()
-	for {
-		if !conn.WaitForStateChange(ctx, conn.GetState()) {
-			return conn, fmt.Errorf("Connection timed out")
-		}
+	// We wait for 1 minute until conn.GetState() is READY.
+	// The interval for this check is 1 second.
+	if err := waitFor(1*time.Minute, 1*time.Second, func() (bool, error) {
 		if conn.GetState() == connectivity.Ready {
-			return conn, nil
+			return false, nil
 		}
+		return true, nil
+	}); err != nil {
+		return nil, fmt.Errorf("Connection timed out")
 	}
+
+	return conn, nil
+}
+
+// f() return <wait as bool, or err>.
+// waitFor stops when f() return (false) or err != nil
+func waitFor(timeout time.Duration, period time.Duration, f func() (bool, error)) error {
```

Comment:
This is great, can we move this function to pkg/util/wait.go and expose it and use it here. There may be other situations we use such a function.

Reply:
:+1: Yep good idea